### PR TITLE
Inheritable Vessel and POI Prototypes

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Makes VesselPrototype and PointOfInterestPrototype inheritable.

Pushes commonly defined items in Vessel and POI prototypes into base parents.

The NFSD outpost is a slightly lighter brown for map readability, the cook role's been removed from the Empress, and engine info for the Tyne and Paladin were added.

## Why / Balance
#2631 wouldn't have been needed with this.

## How to test
Check Tinnia's, make sure the bus stops there (grid has StationTransitComponent) and it can't be moved (linear & angular damping in ShuttleComponent set to 999999).

Check LPBravo, make sure it's hidden.

Purchase regular ships, they should have their IFF label on and appear white.

Purchase pirate ships, they should have their IFF label off and still appear white.

## Media
N/A

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
Existing shuttle PRs could use an update.

**Changelog**
Nothing player facing, no changelog.

### CODE REVIEW

**Comment by whatston3** [comment]
Getting ideas to "paper":
A more holistic approach might be useful here.
If VesselPrototype had inheritance, we could push most of this into parent classes (assuming ComponentRegistry allows it - lessons to learn from EntityPrototype)
Then Menace and friends can inherit from BasePirateShip and have very few unique components.

**Comment by whatston3** [comment]
Might've goofed something up, but the scaffolding's there, and it seems to work okay under very minimal tests.

* NFSD/Courthouse brown is now lighter
* A few vessels missing engine types now have them listed.

Diff's now closer to not adding extra lines, hooray.

File: Resources/Prototypes/_NF/PointsOfInterest/bahamamamas.yml [hunk 1]
```diff
@@ -10,19 +10,14 @@
...
   spawnGroup: RestStop
   gridPath: /Maps/_NF/POI/bahama.yml
   addComponents:
-  - type: IFF
-    color: "#ffa600"
-    readOnly: true
-  - type: Shuttle
-    angularDamping: 999999
-    linearDamping: 999999
   - type: StationTransit
```
Comment (dvir001) [inline Resources/Prototypes/_NF/PointsOfInterest/bahamamamas.yml:22]: Stuff that have a addcomp list don't get overwrite?

Comment (whatston3) [reply Resources/Prototypes/_NF/PointsOfInterest/bahamamamas.yml:22]: It should work like the EntityPrototype, and your ComponentRegistry _does_ accept a list of items.  Should test this better, but I think we're good.

Comment (whatston3) [reply Resources/Prototypes/_NF/PointsOfInterest/bahamamamas.yml:22]: Tested a little bit more, works fine.  LPBravo's hidden, BaseShuttleAntag's fine, all antag ships are hidden with white IFF, and the base ones are all visible with white IFF.

**Review by whatston3 — APPROVED** [review]
Seems fine under test.

Cleaned out the Menace's IFF in-map changes.

Would like a double check that I haven't missed any ships or POIs, but this seems good.

Cook changes to the Empress' job list makes sense.

**Pull request merged**